nansat/mappers/mapper_ascat.py

```diff
@@ -0,0 +1,75 @@
+import re
+import json
+import numpy as np
+from datetime import datetime
+import pythesint as pti
+
+from netCDF4 import Dataset
+
+import gdal
+
+from nansat.vrt import VRT
+from nansat.geolocation import Geolocation
+from nansat.nsr import NSR
+from nansat.domain import Domain
+from nansat.mappers.scatterometers import Mapper as ScatterometryMapper
+from nansat.exceptions import WrongMapperError
+
+class Mapper(ScatterometryMapper):
+    """ Nansat mapper for ASCAT """
+
+    def __init__(self, filename, gdal_dataset, metadata, quartile=0, *args, **kwargs):
+
+        if not 'ascat' in metadata.get('NC_GLOBAL#source', '').lower():
+            raise WrongMapperError
+
+        super(Mapper, self).__init__(filename, gdal_dataset, metadata, quartile=quartile, *args, **kwargs)
+
+        band_lat = self.dataset.GetRasterBand(self._latitude_band_number(gdal_dataset))
+        # Check that it is actually latitudes
+        if not band_lat.GetMetadata()['long_name'] == 'latitude':
+            raise ValueError('Cannot find latitude band')
+        lat = band_lat.ReadAsArray()
+
+        band_lon = self.dataset.GetRasterBand(self._longitude_band_number(gdal_dataset))
+        # Check that it is actually longitudes
+        if not band_lon.GetMetadata()['long_name'] == 'longitude':
+            raise ValueError('Cannot find longitude band')
+        lon = band_lon.ReadAsArray()
+
+        lon = ScatterometryMapper.shift_longitudes(lon)
```

Comment:
There is a jump but this is because nadir is not imaged. 

![ASCAT imaging geometry](https://www.researchgate.net/profile/Yong_Wang260/publication/235553667/figure/fig2/AS:650098526670848@1532006975048/Imaging-geometry-of-ASCAT_W640.jpg)